src/app/calculator/furnaces/o2-enrichment/enrichment-graph/enrichment-graph.component.ts

```diff
@@ -0,0 +1,381 @@
+import { Component, OnInit, Input, ElementRef, ViewChild, HostListener, ChangeDetectorRef, KeyValueDiffers, OnChanges, SimpleChanges, DoCheck } from '@angular/core';
+import { EnrichmentInput, EnrichmentInputData } from '../../../../shared/models/phast/o2Enrichment';
+import { Settings } from '../../../../shared/models/settings';
+import { graphColors } from '../../../../phast/phast-report/report-graphs/graphColors';
+
+import { SimpleChart, TraceCoordinates } from '../../../../shared/models/plotting';
+import * as Plotly from 'plotly.js';
+import { O2EnrichmentService, DisplayPoint } from '../o2-enrichment.service';
+import { Subscription } from 'rxjs';
+
+@Component({
+  selector: 'app-enrichment-graph',
+  templateUrl: './enrichment-graph.component.html',
+  styleUrls: ['./enrichment-graph.component.css']
+})
+export class EnrichmentGraphComponent implements OnInit {
+
+  @Input()
+  settings: Settings;
+
+  // DOM
+  @ViewChild("ngChartContainer", { static: false }) ngChartContainer: ElementRef;
+  @ViewChild('dataSummaryTable', { static: false }) dataSummaryTable: ElementRef;
+  dataSummaryTableString: any;
+  tabPanelChartId: string = 'tabPanelDiv';
+  expandedChartId: string = 'expandedChartDiv';
+  currentChartId: string = 'tabPanelDiv';
+
+  // Graph Data
+  selectedDataPoints: Array<DisplayPoint>;
+  defaultSelectedPointsData: Array<DisplayPoint>;
+  pointColors: Array<string>;
+  enrichmentChart: SimpleChart;
+  enrichmentInputsSub: Subscription;
+  enrichmentInputs: Array<EnrichmentInput>;
+
+  // Trace Defaults
+  defaultPointOutlineColor = 'rgba(0, 0, 0, .6)';
+  defaultPointColor = 'rgba(0, 0, 0, 0)';
+  defaultTraceCount: number = 1;
+
+  // Tooltips
+  hoverBtnGridLines: boolean = false;
+  displayGridLinesTooltip: boolean = false;
+  hoverBtnExpand: boolean = false;
+  displayExpandTooltip: boolean = false;
+  hoverBtnCollapse: boolean = false;
+  displayCollapseTooltip: boolean = false;
+  expanded: boolean = false;
+
+  initResizeCompleted: boolean = false;
+
+  @HostListener('document:keyup', ['$event'])
+  closeExpandedGraph(event) {
+    if (this.expanded) {
+      if (event.code === 'Escape') {
+        this.contractChart();
+      }
+    }
+  }
+
+  constructor(private o2EnrichmentService: O2EnrichmentService,
+    private cd: ChangeDetectorRef,
+  ) {
+  }
+
+  ngOnInit() {
+    this.pointColors = graphColors;
+    this.triggerInitialResize();
+    this.initSubscriptions();
+  }
+
+  ngOnDestroy() {
+    this.enrichmentInputsSub.unsubscribe();
+  }
+
+  initSubscriptions() {
+    this.enrichmentInputsSub = this.o2EnrichmentService.enrichmentInputs.subscribe(inputs => {
+      if (inputs && this.initResizeCompleted) {
```

Comment:
No longer needed. This was a way for me to prevent duplicate traces on init from tab switching (when graph object was still populating with empty trace x,y)